frontend/src/app/main/ui/workspace/colorpicker/ramp.cljs

```diff
@@ -63,38 +64,52 @@
            :h h :s s :v v
            :r r :g g :b b)))
 
-(mf/defc ramp-selector
+(mf/defc ramp-selector*
   [{:keys [color disable-opacity on-change on-start-drag on-finish-drag]}]
-  (let [internal-color
+  (let [internal-color*
         (mf/use-state #(enrich-color-map color))
 
-        {:keys [h s v hex alpha]} @internal-color
+        internal-color
+        (deref internal-color*)
+
+        h     (get internal-color :h)
+        s     (get internal-color :s)
+        v     (get internal-color :v)
+        hex   (get internal-color :hex)
+        alpha (get internal-color :alpha)
+
+        bullet-color
+        (mf/with-memo [hex alpha]
+          {:color hex :opacity alpha})
+
         on-change-value-saturation
-        (fn [new-saturation new-value]
-          (let [new-color (-> @internal-color
-                              (assoc :s new-saturation)
-                              (assoc :v new-value)
-                              (enrich-color-map))]
-            (reset! internal-color new-color)
-            (on-change new-color)))
+        (mf/use-fn
+         (mf/deps internal-color on-change)
```

Comment:
It's important to cache all internal functions in a `mf/use-fn` to avoid creating a new function each time.